src/gui/sdlmain.cpp

```diff
@@ -1308,38 +1391,48 @@ static SDL_Window *SetupWindowScaled(SCREEN_TYPES screen_type, bool resizable)
 	if (sdl.scaling_mode == SCALING_MODE::PERFECT)
 		return setup_window_pp(screen_type, resizable);
 
-	int w, h;
+	int window_width;
+	int window_height;
 	if (sdl.desktop.fullscreen) {
-		w = sdl.desktop.full.fixed ? sdl.desktop.full.width : 0;
-		h = sdl.desktop.full.fixed ? sdl.desktop.full.height : 0;
+		window_width = sdl.desktop.full.fixed ? sdl.desktop.full.width : 0;
+		window_height = sdl.desktop.full.fixed ? sdl.desktop.full.height : 0;
 	} else {
-		w = sdl.desktop.window.width;
-		h = sdl.desktop.window.height;
+		window_width  = sdl.desktop.window.width;
+		window_height = sdl.desktop.window.height;
 	}
 
-	if (w > 0 && h > 0) {
-		sdl.clip = calc_viewport_fit(w, h);
-		sdl.window = SetWindowMode(screen_type, w, h,
-		                           sdl.desktop.fullscreen, resizable);
+	if (window_width > 0 && window_height > 0) {
+		sdl.window = SetWindowMode(screen_type,
+		                           window_width,
+		                           window_height,
+		                           sdl.desktop.fullscreen,
+		                           resizable);
+
+		const auto canvas_dims = get_canvas_size(screen_type);
+		assert(canvas_dims.x > 0 && canvas_dims.y > 0);
+
+		sdl.clip = calc_viewport_fit(canvas_dims.x, canvas_dims.y);
 
 		const auto is_window_fullscreen = SDL_GetWindowFlags(sdl.window) &
 		                                  SDL_WINDOW_FULLSCREEN;
 		if (sdl.window && is_window_fullscreen) {
-			int win_width;
-			SDL_GetWindowSize(sdl.window, &win_width, NULL);
-			sdl.clip.x = (win_width - sdl.clip.w) / 2;
-			sdl.clip.y = (h - sdl.clip.h) / 2;
+			sdl.clip.x = (canvas_dims.x - sdl.clip.w) / 2;
+			sdl.clip.y = (canvas_dims.y - sdl.clip.h) / 2;
```

Comment:
When I see the mixing of `x, y` and `w, h`, I now see that my recommendation to use an `SDL_Point` for `get_canvas_size()` worsened the readability here. Sorry for that.

I think the right approach now, is to make `get_canvas_size` return an `SDL_Rect` type, which has `.w` and `.h` integer members.

I had previously recommended using the name `canvas_dims` to clarify that its .x and .y SDL_Point members are dimensions and not coordinates. But when we switch to the SDL_Rect, that clarification isn't need.

So code can get much clearer here..

 ``` c++
const auto canvas = get_canvas_size(screen_type);
assert(canvas.w > 0 && canvas.h > 0);

sdl.clip = calc_viewport_fit(canvas.w, canvas.h);
...
```

Reply:
And empty x and y members of the struct? Just disregard them?

Reply:
SDL_Rect is a decent fit: we're dealing with a rectangle having given dimensions, but we're not concerned with the rectangles position.

But if we're worried about "wasting" those unused two coordinates, the alternative is creating our rectangle struct/class without the ability to place it in space.

I'm OK with either approach because I feel that carrying around dimensions together is a better than them being free-floating integers.

Reply:
For the scope of the PR, I think `SDL_Rect` is a local maximum.

If we want to create a new dimension-object, then I would suggest that effort go into a new PR which overhauls all of sdlmain with it, doing away with the current grab-bag mix of SDL_Point, SDL_Rect, and free-floating integers.

Reply:
Replaced with `SDL_Rect`.

Reply:
Thanks @GranMinigun (definitely easier on the eyes, there!).  Resolving.